python/pyarrow/_flight.pyx

```diff
@@ -844,7 +844,7 @@ cdef class FlightInfo(_Weakrefable):
         return obj
 
     def __init__(self, Schema schema, FlightDescriptor descriptor, endpoints,
-                 total_records, total_bytes):
+                 total_records, total_bytes, ordered, app_metadata):
```

Comment:
```suggestion
                 total_records, total_bytes, ordered=False, app_metadata=b""):
```

These new parameters should be optional for backwards compatibility

Reply:
Tried that, this is not supported by Cython :disappointed:.

Reply:
That's odd, there are other `cdef` classes in this same file that have optional parameters in their `__init__`, eg. `FlightError` also has an optional `bytes` parameter. What is the error you get?

Reply:
Looks like it works this time. :partying_face: